### ISSUE TITLE
Z Index support

### ISSUE THREAD

**Issue by cromerc** [mapeditor/tiled-to-godot-export#23]
It would be nice to be able to adjust the Z Index on specific tiles from the tiled editor. This would help with tiles that should cover other objects. As an example, the game I am working on has a doorway, the player sprite should walk under the doorway, but right now it walks on top of the doorway.

I am able to achieve this result manually by modifying the tileset that gets exported and setting the specific tile to have a Z Index of 1. However if I reexport the project this gets overridden and I have to redo all the Z Indexes in the tileset. The ideal solution would be to allow a custom property on a tile called "zIndex" which should be an integer, this should get exported to the godot tileset format.

Here is an example godot tileset that uses the z index:
```
[gd_resource type="TileSet" load_steps=5 format=2]

[ext_resource path="res://maps/dungeon.png" type="Texture" id=1]

[sub_resource type="ConvexPolygonShape2D" id=1]
points = PoolVector2Array( 9, 0, 54, 0, 54, 54, 9, 54 )

[sub_resource type="ConvexPolygonShape2D" id=2]
points = PoolVector2Array( 6, 0, 54.5, 0, 54.5, 53.5, 6, 53.5 )

[sub_resource type="ConvexPolygonShape2D" id=3]
points = PoolVector2Array( 6, 0, 54.5, 0, 54.5, 53.5, 6, 53.5 )

[resource]
0/name = "dungeon 0"
0/texture = ExtResource( 1 )
0/tex_offset = Vector2( 0, 0 )
0/modulate = Color( 1, 1, 1, 1 )
0/region = Rect2( 0, 0, 1024, 1024 )
0/tile_mode = 2
0/autotile/icon_coordinate = Vector2( 0, 0 )
0/autotile/tile_size = Vector2( 64, 64 )
0/autotile/spacing = 0
0/autotile/occluder_map = [  ]
0/autotile/navpoly_map = [  ]
0/autotile/priority_map = [  ]
0/autotile/z_index_map = [ Vector3( 3, 9, 1 ) ]
0/occluder_offset = Vector2( 0, 0 )
0/navigation_offset = Vector2( 0, 0 )
0/shape_offset = Vector2( 0, 0 )
0/shape_transform = Transform2D( 1, 0, 0, 1, 0, 0 )
0/shape = SubResource( 1 )
0/shape_one_way = false
0/shape_one_way_margin = 1.0
0/shapes = [ {
"autotile_coord": Vector2( 14, 1 ),
"one_way": false,
"one_way_margin": 1.0,
"shape": SubResource( 1 ),
"shape_transform": Transform2D( 1, 0, 0, 1, 0, 0 )
}, {
"autotile_coord": Vector2( 13, 12 ),
"one_way": false,
"one_way_margin": 1.0,
"shape": SubResource( 2 ),
"shape_transform": Transform2D( 1, 0, 0, 1, 0, 0 )
}, {
"autotile_coord": Vector2( 9, 15 ),
"one_way": false,
"one_way_margin": 1.0,
"shape": SubResource( 3 ),
"shape_transform": Transform2D( 1, 0, 0, 1, 0, 0 )
} ]
0/z_index = 0
```
The important line to generate is:
0/autotile/z_index_map = [ Vector3( 3, 9, 1 ) ]

This indicates that the tile located at x = 3 and y = 9 is set to a Z Index of 1. This means the player passes under that tile because the player I have is set to Z Index 0.

**Comment by cromerc** [comment]
I was looking at the forks of this project just now and noticed that somebody did something similar, they made it so that layers have their z-index exported to the godot tileset. Here is the commit: https://github.com/agmcleod/tiled-to-godot-export/commit/e0e3ae38f46c4e8216cb0c73564ffc7bf05bba74

This does allow a z-index for a whole layer to be exported, so letting and individual tile z-index be exported would still be missing.

It would be nice to have both forms as an option when using this plugin because I don't necessarily want to create a whole separate layer just to have a new z-index on a few specific tiles(doorways).

**Comment by bjorn** [comment]
@cromerc It sounds like a nice feature to add. Would you be interested in having a go at implementing it? Also, I guess the final line `0/z_index = 0` sets the z-index for all tiles in the tileset, which don't specify an override? Might be nice to support as well, since it can help to reduce the file size.

@agmcleod I think your change to support setting the z-index on each layer would be nice to merge as well. Would you like to open a pull request for it? A small conflict with change 3c2848d03198550fa7ac6634171a4fb700267c10 will need to be resolved.

**Comment by agmcleod** [comment]
I'll see if I can get one up later today. I was holding onto it while my other PR was open, but didn't have time to make one since that got merged :D

**Comment by cromerc** [comment]
> @cromerc It sounds like a nice feature to add. Would you be interested in having a go at implementing it?
If I had more time, I would probably look into doing it. But at the moment I have very little time to spare.

**Comment by MikeMnD** [comment]
So with the PR merged we can close this.

**Comment by bjorn** [comment]
@MikeMnD Hmm, not sure about that, since the PR only adds support for the per-layer Z-index, whereas the request here was for per-tile Z-index. Both seem to be supported by Godot.

**Comment by bjorn** [comment]
Closed by #44.
